netconf.c

```diff
@@ -737,12 +794,67 @@ handle_unlock (struct netconf_session *session, xmlNode * rpc)
     }
 
     /* Unlock running datastore */
-    _reset_lock ();
+    reset_lock ();
 
     /* Success */
     return send_rpc_ok (session, rpc, false);
 }
 
+static bool
+handle_kill_session (struct netconf_session *session, xmlNode * rpc)
+{
+    xmlNode *action = xmlFirstElementChild (rpc);
+    xmlNode *node;
+    uint32_t kill_session_id = 0;
+    char *msg = NULL;
+    struct netconf_session *kill_session = NULL;
+
+
+    /* Validate request */
+    node = xmlFindNodeByName (action, BAD_CAST "session-id");
+    if (!node)
+    {
+        VERBOSE ("Missing \"session-id\" element");
+        msg = "Missing \"session-id\" element";
+        return send_rpc_error (session, rpc, "missing-element", msg, NULL);
+    }
+
+    /* Return an "invalid-error" if the request is made by the current session */
+    sscanf ((char *) xmlNodeGetContent (node), "%u", &kill_session_id);
```

Comment:
Does this return a copy or just a pointer into existing memory?

Reply:
```
/**
 * xmlNodeGetContent:
 * @cur:  the node being read
 *
 * Read the value of a node, this can be either the text carried
 * directly by this node if it's a TEXT node or the aggregate string
 * of the values carried by this node child's (TEXT and ENTITY_REF).
 * Entity references are substituted.
 * Returns a new #xmlChar * or NULL if no content is available.
 *     It's up to the caller to free the memory with xmlFree().
 */
```
Will need to free the memory. Let me fix this.